pallets/credit/src/lib.rs

```diff
@@ -698,18 +697,23 @@ pub mod pallet {
             Self::do_unstaking_slash_credit(&user)
         }
 
-        #[pallet::weight(10_000 + T::DbWeight::get().reads_writes(0,1))]
-        pub fn set_credit_admin(origin: OriginFor<T>, admin: T::AccountId) -> DispatchResult {
-            ensure_root(origin)?;
-            CreditAdmin::<T>::put(admin.clone());
-            Self::deposit_event(Event::SetAdmin(admin));
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::add_or_update_credit_data())]
+        pub fn external_set_credit_data(
```

Comment:
Not test covered. I think it's better to also add benchmark and weights, so in the future, just delete the `add_or_update_credit_data` related code

Reply:
same logic as add_or_update_credit_data. when add_or_update_credit_data deprecated,  beachmark function name maybe changed